### PR TITLE
[ISSUE #7039] Temporary fix encryptedContent bug.

### PR DESCRIPTION
**Pull request by CherishCai**
## What is the purpose of the change

fix(#7039)

## Brief changelog

不是好的实现方式，临时修复至配置解密可用。期待社区统一考虑解密 Filter，重构这块逻辑。不过也先可用再优化吧。。。。

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [x] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by CherishCai** [comment]
或者后面有空，协商一下整体逻辑重构一波，cache 缓存在内存的配置内从是解密后的，应该就没问题。

**Comment by KomachiSion** [comment]
@li-xiao-shuang @realJackSun @shiyiyue1102 Please review this PR.

**Comment by CherishCai** [comment]
@li-xiao-shuang @realJackSun @shiyiyue1102 
是否讨论下呢，先可用。我这边内部定制版是验证并发布了 2.x 的配置加解密。

**Comment by CherishCai** [comment]
naming CI fail, Doesn't seem to be related to this PR change.
image

**Review by shiyiyue1102 — APPROVED** [review]

File: client/src/main/java/com/alibaba/nacos/client/config/impl/CacheData.java [hunk 1]
```diff
@@ -317,11 +318,22 @@ private void safeNotifyListener(final String dataId, final String group, final S
                 listener.receiveConfigInfo(contentTmp);
                 // compare lastContent and content
                 if (listener instanceof AbstractConfigChangeListener) {
-                    Map data = ConfigChangeHandler.getInstance()
-                            .parseChangeData(listenerWrap.lastContent, content, type);
```
[the remark this answers is not in the document]
Comment (CherishCai) [reply client/src/main/java/com/alibaba/nacos/client/config/impl/CacheData.java:321]: if content be encrypted,  
```java
configFilterChainManager.doFilter(null, cr);
String contentTmp = cr.getContent();
```
contentTmp will get a decryptContent,  So ConfigChangeHandler.getInstance().parseChangeData should be use it.

[the remark this answers is not in the document]
Comment (CherishCai) [reply client/src/main/java/com/alibaba/nacos/client/config/impl/CacheData.java:321]: and then `listenerWrap.lastContent = contentTmp;` hold a decryptContent for next compare

**Review by li-xiao-shuang — APPROVED** [review]

**Pull request merged**